src/components/Markings/components/SvgOverlay.js

```diff
@@ -0,0 +1,177 @@
+            onMoveShouldSetPanResponderCapture: () => {return this.props.mode === 'draw'},
+            onPanResponderGrant: (evt) => {
+                const { locationX, locationY } = evt.nativeEvent
+                this.setState({
+                    isDrawing: true,
+                    previewSquareX: locationX - INITIAL_SQUARE_SIDE,
+                    previewSquareY: locationY - INITIAL_SQUARE_SIDE
+                })
+            },
+            onPanResponderMove: (evt, gestureState) => {
+                const { dx, dy } = gestureState
+                this.widthAnimated.setValue(INITIAL_SQUARE_SIDE + dx)
+                this.heightAnimated.setValue(INITIAL_SQUARE_SIDE + dy)
+            },
+            onPanResponderTerminationRequest: () => true,
+            onPanResponderRelease: (evt, gestureState) => {
+                const { previewSquareX, previewSquareY } = this.state
+                const { dx, dy } = gestureState
+                const shape = {
+                    type: 'square',
+                    color: this.props.color,
+                    x: previewSquareX,
+                    y: previewSquareY,
+                    width: INITIAL_SQUARE_SIDE + dx,
+                    height: INITIAL_SQUARE_SIDE + dy
+                }
+
+                this.setState({ isDrawing: false })
+                this.props.drawingScreenActions.addShape(shape)
+            },
+            onPanResponderTerminate: () => {
+                this.setState({
+                    squareWidth: INITIAL_SQUARE_SIDE,
+                    squareHeight: INITIAL_SQUARE_SIDE,
+                    isDrawing: false,
+                })
+
+
+            },
+            onShouldBlockNativeResponder: () => true
+          });
+    }
+
+    renderShapes() {
+        const shapeArray = []
+        const convertObjectToComponent = (shape, index) => {
+            const { type } = shape
+            switch (type) {
+                case ('square'):
+                    shapeArray.push(
+                        <DragableSquare 
+                            key={index} 
+                            index={index}
+                            isEditable={this.props.mode === 'edit'}
+                            isDeletable={this.props.mode === 'erase'}
+                            onDelete={() => {
+                                this.props.drawingScreenActions.removeShapeAtIndex(index)
+                            }}
+                        />
+                    )
+            }
+        }
+        
+        R.mapObjIndexed(convertObjectToComponent, this.props.shapes)
+        return shapeArray
+    }
+
+    render() {
+        return (
+            <Svg
+                { ...this.panResponder.panHandlers }
+                style={styles.svg}
+                preserveAspectRatio="xMidYMid meet"
+            >
+                { this.state.isDrawing ? 
+                    <Rect 
```

Comment:
Will this module likely need some rewriting in the future as more shapes are introduced? I noticed in the `DrawingTitle` component you have a case to determine what the shape is, where rect is currently the only option.

Reply:
Yeah, I'll add something here to make it more modular